earth_enterprise/src/fusion/autoingest/AssetFactory_unittest.cpp

```diff
@@ -51,25 +61,106 @@ class MockAssetStorage
               }
 };
 
+
+// added ID field to aid in comparison
 class MockAssetConfig
 {
 public:
     int8_t ID;
-    MockAssetConfig(int8 _ID = -1) : ID(_ID){}
+    MockAssetConfig(int8 _ID = -1) : ID(_ID) {}
     bool operator==(const MockAssetConfig& other) const { return ID == other.ID; }
     bool operator==(uint8_t _ID) { return ID == _ID; }
     void operator=(const MockAssetConfig& other) { ID = other.ID; }
     void operator=(uint8_t _ID) { ID = _ID; }
  };
 
+class MockAssetVersionImpl : public MockAssetStorage
+{
+public:
+    using MutableAssetType = MockMutableAsset;
+    static AssetDefs::Type EXPECTED_TYPE;
+    static string EXPECTED_SUBTYPE;
+
+    using Base = MockAssetStorage;
+
+    MockAssetConfig config;
+    bool needed;
+
+    void Modify(const khMetaData& _meta, const MockAssetConfig& _config)
+    {
+        meta = _meta;
+        config = _config;
+    }
+
+    void Modify(const vector<SharedString>& _inputs,
+                const khMetaData& _meta,
+                const MockAssetConfig& _config)
+    {
+        inputs = _inputs;
+        Modify(_meta, _config);
+    }
+
+    MockAssetVersionImpl(MockAssetStorage storage,
+                  const MockAssetConfig &config_)
+                : MockAssetStorage(storage), config(config_), needed(false) {}
+};
+
+
+class MockMutableAssetVersion : public MockAssetStorage
+{
+public:
+    using Impl = MockAssetVersionImpl;
+    static string testSubTypeToUseForStringConstructor;
+    shared_ptr<Impl> impl = nullptr;
+
+    MockMutableAssetVersion() = default;
+    MockMutableAssetVersion(const string& ref_)
+    {
+        MockAssetStorage storage;
+        storage.name = ref_;
+        storage.type = Impl::EXPECTED_TYPE;
+        storage.subtype = testSubTypeToUseForStringConstructor;
+
+        MockAssetConfig config;
+        impl = make_shared<Impl>(storage, config);
+    }
+
+    MockMutableAssetVersion(const MockAssetStorage& storage, const MockAssetConfig& config)
+    {
+        impl = make_shared<Impl>(storage, config);
+    }
+
+    MockMutableAssetVersion(const string& name, AssetDefs::Type type)
+    {
+        MockAssetStorage storage;
+        storage.name = name;
+        storage.type = type;
+        MockAssetConfig config;
+        impl = make_shared<Impl>(storage,config);
+    }
+
+    explicit operator bool(void) const
+    {
+        return impl != nullptr;
+    }
+
+    Impl* operator->() const
+    {
+        return impl.get();
+    }
+
+};
+
 class MockAssetImpl: public MockAssetStorage
 {
 public:
   static AssetDefs::Type EXPECTED_TYPE;
   static string EXPECTED_SUBTYPE;
 
   using Base = MockAssetStorage;
+  using Impl = MockAssetImpl;
```

Comment:
`using Impl = MockAssetImpl;` seems out of place here since it's just the type of this class.